internal/cpp/opencc/utils.c

```
@@ -23,8 +23,10 @@ void perr(const char *str) { fputs(str, stderr); }
 int qsort_int_cmp(const void *a, const void *b) { return *((int *)a) - *((int *)b); }
 
 char *mstrcpy(const char *str) {
-    char *strbuf = (char *)malloc(sizeof(char) * (strlen(str) + 1));
-    strcpy(strbuf, str);
+    size_t len = strlen(str);
+    char *strbuf = (char *)malloc(sizeof(char) * (len + 1));
+    strncpy(strbuf, str, len);
```

Cppcheck (2.20.0)

[warning] If memory allocation fails, then there is a possible null pointer dereference

(nullPointerOutOfMemory)